src/SBInclinedExponential.cpp

```diff
@@ -0,0 +1,367 @@
+        oss.precision(std::numeric_limits<double>::digits10 + 4);
+        oss << "galsim._galsim.SBInclinedExponential("<<getI()<<", "<<getScaleRadius()<<", "<<getScaleHeight();
+        oss <<", "<<getFlux()<<", False";
+        oss << ", galsim.GSParams("<<*gsparams<<"))";
+        return oss.str();
+    }
+
+    LRUCache< boost::tuple<double, GSParamsPtr >, InclinedExponentialInfo >
+        SBInclinedExponential::SBInclinedExponentialImpl::cache(sbp::max_inclined_exponential_cache);
+
+    SBInclinedExponential::SBInclinedExponentialImpl::SBInclinedExponentialImpl(double i, double scale_radius, double scale_height, double flux,
+            const GSParamsPtr& gsparams) :
+        SBProfileImpl(gsparams),
+        _r0(scale_radius),
+        _h0(scale_height),
+        _flux(flux),
+        _inv_r0(1./_r0),
+        _h_tani_over_r(scale_height*std::tan(i)/scale_radius),
+        _r0_cosi(_r0*std::cos(i)),
+        _inv_r0_cosi(1./(scale_radius*std::cos(i))),
+        // Start with InclinedExponentialInfo regardless of value of trunc
+        _info(cache.get(boost::make_tuple(_h_tani_over_r, this->gsparams.duplicate())))
+    {
+        dbg<<"Start SBInclinedExponential constructor:\n";
+        dbg<<"i = "<<_i<<std::endl;
+        dbg<<"scale radius = "<<_r0<<std::endl;
+        dbg<<"scale height = "<<_h0<<std::endl;
+        dbg<<"flux = "<<_flux<<std::endl;
+
+        // Ensure i is positive and in the range 0 <= i <= M_PI/2
+
+        // Work with positive values first
+        if(i<0) i = -i;
+
+        // Get into the proper range
+        if(i>M_PI/2.)
+        {
+            i -= int(i/(2*M_PI))*2*M_PI;
+            if(i<0) i = -i; // Since we might have gone below zero again
+        }
+
+        // Check if i is within allowed limits, and institute special handling if it isn't
+        if(i>sbp::maximum_i)
+        {
+            // Perfectly edge-on isn't analytic, so we truncate at the maximum value
+            _i = sbp::maximum_i;
+        }
+        else
+        {
+            // Normal setup here
+            _i = i;
+            _h_tani_over_r = scale_height*std::tan(i)/scale_radius;
+            _r0_cosi = _r0*std::cos(i);
+            _inv_r0_cosi = 1./(scale_radius*std::cos(i));
+        }
+
+        _info = boost::shared_ptr<InclinedExponentialInfo>(cache.get(boost::make_tuple(_h_tani_over_r, this->gsparams.duplicate())));
+
+        xdbg<<"_h_tani_over_r = "<<_h_tani_over_r<<std::endl;
+        xdbg<<"_r0_cosi = "<<_r0_cosi<<std::endl;
+
+        /* Shooting NYI
+        _shootnorm = _flux * _info->getXNorm(); // For shooting, we don't need the 1/r0^2 factor.
+        _xnorm = _shootnorm * _inv_r0 * _inv_r0 ;
+        dbg<<"norms = "<<_xnorm<<", "<<_shootnorm<<std::endl;
+        */
+    }
+
+    double SBInclinedExponential::SBInclinedExponentialImpl::xValue(const Position<double>& p) const
+    {
+        throw std::logic_error("Real-space expression. of SBInclinedExponential NYI.");
```

Comment:
This isn't a logic error.  Probably a std::runtime_error would be appropriate.  At least until we do something about issue #755.